Microsoft.Azure.TypeEdge/Modules/TypeModule.cs

```diff
@@ -144,7 +151,7 @@ internal virtual InitializationResult _Init(IConfigurationRoot configuration, IC
             if (disableSslCertificateValidationKey)
                 settings.RemoteCertificateValidationCallback = (sender, certificate, chain, sslPolicyErrors) =>
                     true;
-            _transportSettings = new ITransportSettings[] {settings};
+            TransportSettings = new ITransportSettings[] { settings };
```

Comment:
Whats your intention regarding how a user could override the transport settings for this module?

Reply:
The idea is that the Module base class (TypeModule) will set a default value on the TransportSettings, and everyone that inherits from this class will be able to change these settings.

If you take a look [here](https://github.com/Azure/TypeEdge/blob/2bf51c72716815d5a41224c68c2fc7a9f807ee5e/Microsoft.Azure.TypeEdge/Modules/TypeModule.cs#L141-L156) the _Init eventually calls the Init(), which is virtual and can be overridden.  If overridden, this is the opportunity to change the default settings of the transport.

The problem lies in the fact that the transport settings are not reflected by the manifest, and because of that, anything set by the emulator as code will be lost when running the modules inside a container as a typical edge module. This is the reason why I think the customization of the transport settings should be set inside every module, by reading the module settings.

Lmk what you think.

Reply:
Ah ok - didn't see that. Looks fine now that I know.

I didn't even know that individual modules are able to specifiy their custom transport settings, since the connection to the cloud side IoT Hub is maintained by the edge agent / edge hub? Or are the module specific transport settings used only for the communication from the module to the local edge agent / hub? If so, how could we specify the transport settings used by the edge agent / hub to connect to the cloud side IoT Hub using TypeEdge? I'm asking this, because especially during development time its sometimes not possible to work with AMQP connections because some corporate LANS block required ports.

Reply:
Correct, the module transport settings are for the communication channel between the module and the hub, in other words, you could have modules with different settings in the same edge app. This has to be controlled in a per-module mode, because this configuration does not formally exist in the manifest. If reading the config is properly implemented in the module (as it is in TypeEdge) this can be overridden by the module [createOptions](https://github.com/Azure/TypeEdge/blob/b04554d700b30a02b725bac315f4903a0ef5c6c2/Examples/QuickStart/Thermostat.Emulator/manifest.json#L39) section of the manifest.

In the same way, the EdgeHub has some default values that can be overridden. These default values are defined [here](https://github.com/Azure/TypeEdge/blob/master/Microsoft.Azure.TypeEdge.Host/appsettings_hub.json). This file comes with the Host nuget package when you implement your Emulator.

Completing this PR, I'll need more time to publish the new packages, but your should be able to  produce unsigned packages to continue your work, let me know otherwise